docs/src/rules/require-unicode-regexp.md

```diff
@@ -58,6 +58,24 @@ Therefore, the `u` and `v` flags let us work better with regular expressions.
 
 This rule aims to enforce the use of `u` or `v` flag on regular expressions.
 
+### Exception: When `i` flag and `\w` are used together
+
+If a regular expression uses both the `i` flag and the `\w` character class, adding the `u` flag may change its behavior due to Unicode case folding rules.
+
+For example:
+
+```js
+const regexWithoutU = /^\w+$/i;
+const regexWithU = /^\w+$/iu;
+
+const str = "\u017f\u212a"; // Example Unicode characters
+
+console.log(regexWithoutU.test(str)); // true
+console.log(regexWithU.test(str)); // false
+```
+
+In such cases, this rule does not enforce adding the u flag, as it can lead to unintended changes in pattern matching.
+
```

Comment:
This sounds like the rule has a different behavior in these cases, but it doesn't. Since this isn't an exception in this rule, it might be better to add this note as a subsection of "When Not To Use It" and advise using an `eslint-disable` comment.